docs/development/modules/themes/creating-themes.md

```diff
@@ -0,0 +1,126 @@
+
+### Using Base Theme Options
+The constructor of a theme class allows Dependency Injection. This means that we can inject the Current Theme Object, use existing options for our theme. It allows us to use the base theme to style our Manialink, in order to keep a consistent style.
+
+You may be wondering, can't we just use the base theme options directly instead of creating a theme with new options that uses them? This is where the philosophy of the controller comes in. Because we wan't provide people with a highly customizable and versatile server controller so that they can make it exactly how they want. So everything within a Manialink has it's own option, which can be overriden in the config or by other Modules to modify the look of it.
+
+So now that you understand why we do it this way, let's see an example that uses some of the base options:
+
+```cs
+[Theme(Name = "MyTheme", Description = "This is my custom theme.")]
+public class MyTheme : Theme<MyTheme>
+{
+    private readonly dynamic _theme;
+
+    public MyTheme(IThemeManager theme) => _theme = theme.Theme;
+
+    public override Task ConfigureAsync()
+    {
+        Set("MyModule.MyManialink.Text").To(_theme.UI_TextPrimary);
+        Set("MyModule.MyManialink.Bg").To(_theme.UI_BgPrimary);
+
+        return Task.CompletedTask;
+    }
+}
+```
+
+### Replacing Existing Manialink Components
+It is also possible to replace existing Manialink components with new ones. So let's say you implemented a custom Button and want the entire UI to use that button instead. You can then use the `Replace(OldComponentName).With(NewComponentName)` pattern:
+
+```cs
+[Theme(Name = "MyTheme", Description = "This is my custom theme.")]
+public class MyTheme : Theme<MyTheme>
+{
+    public override Task ConfigureAsync()
+    {
+        Replace("UI.Button").With("MyModule.MyButton");
+
+        return Task.CompletedTask;
+    }
+}
+```
+
+The way this work now, is that whenever `UI.Button` is imported or referenced in some way, it will actually use `MyModule.MyButton` instead.
+
+### Overriding Existing Themes
+A theme can also override existing themes using the `Override` option in the `[Theme]` attribute.
+
+So let's say we want to override the theme for the Button component. We can then do the following:
+
+```cs
+[Theme(Name = "NewButtonTheme", Description = "New theme for the button.", Override = typeof(DefaultButtonTheme))]
+public class MyTheme : DefaultButtonTheme
+{
+    public override async Task ConfigureAsync()
+    {
+        await base.ConfigureAsync();
+
+        Set("UI.Button.Default.Bg").To("0000FF"); // set button background to blue
+    }
+}
+```
+
+This shows another feature of the Theme Framework. We can inherit other themes. This is very useful if you just want to replace certain options, and not having to re-define every single option the original theme provides.
```

Comment:
```suggestion
This shows another feature of the Theme framework. We can inherit other themes. This is very useful if you just want to replace certain options without having to re-define every single option the original theme provides.
```

Reply:
"Theme Framework" is inteded, thats the name of it.